### ISSUE TITLE
xpdf is broken on darwin

### ISSUE THREAD

**Issue by veprbl** [NixOS/nixpkgs#34907]
## Issue description

```
CMake Error in xpdf-qt/CMakeLists.txt:
  No known features for CXX compiler

  "Clang"

  version 4.0.1.
```

First breakage by e42c62e05f47c221a393722456d97cac8043df94

https://hydra.nixos.org/job/nixpkgs/trunk/xpdf.x86_64-darwin/all?page=3
Last success 2017-08-21

### Steps to reproduce

```
nix-shell -p xpdf
```

## Technical details

 - system: `"x86_64-darwin"`
 - host os: `Darwin 16.7.0, macOS 10.12.6`
 - multi-user?: `no`
 - sandbox: `no`
 - version: `nix-env (Nix) 1.11.15`
 - channels: `"nixpkgs-17.03pre92678.1bd4c08"` <- current master

cc @Mic92

**Comment by Mic92** [comment]
I suppose qt should be disabled on macOS then.

**Comment by Kaali** [comment]
I'm just now solving a similar issue with wireshark. CMake version has to be quite high, and it has to be defined even before the project is defined. Similar patch was done here: https://github.com/NixOS/nixpkgs/commit/b693f989f65d55d2213c52b716cf26fe60e40ad0

**Comment by Mic92** [comment]
Sounds like a trivial patch, that can be also upstreamed as well.

**Comment by veprbl** [comment]
Thanks @Kaali ! I was able to implement a fix #36177

I still have one problem with missing icons:
![missing icons](http://aspid.ru/~dump/xpdf.png)

**Comment by veprbl** [comment]
#36188
